src/nedap/utils/spec/impl/check.cljc

```diff
@@ -35,7 +35,14 @@
                                                                                  (pr-str spec-quoted#)
                                                                                  "\n\n-------------------------"))
                     true                      println)
-                  (throw (ex-info "Validation failed" {:spec         spec-quoted#
-                                                       :faulty-value x-quoted#
-                                                       :explanation  (~explain spec# x#)})))))
+                  (throw (ex-info "Validation failed"
+                                  ;; :spec and :faulty-value are legacy keys without strong associated semantics.
+                                  ;; However programs may depend strongly on them. Please don't remove them.
+                                  {:spec                spec-quoted#
+                                   :spec-object         spec#
+                                   :quoted-spec         spec-quoted#
```

Comment:
As the comment suggests, `spec` and `faulty-value` are legacy keys; ideally I'd just rm them but I think we're all alergic to gratuitous breaking changes :)

The new keys state more clearly whether something is "quoted" (a symbol which is bound to an object) or the underyling literal object

Reply:
What about just sticking with `spec` an `faulty-value` ? providing the same keys might be a problem because we need to keep the value in sync.

Reply:
>  keep the value in sync.

Can you expand on that?

(by intuition it doesn't seem a problem to me: given a binding `(let [x ...]))`, you can repeat `[x x x x x]` as many times as you want, without fear of 'sync problems')

Also, this PR is used already in speced.def 2.0.0, so I'm not sure it's worth to introduce any churn or source of confusion.

Reply:
> you can repeat [x x x x x] as many times as you want, without fear of 'sync problems')

yes it's trivial to repeat, but it is a manual thing to keep in mind.
Using extra-descriptive keywords for the same value might be confusing, especially because you can't mark them as deprecated (right?).

This might be fixable with a test though (if we don't already cover explicit equality between `spec` == `quoted-spec`)

Reply:
I can give a revamp to the documentation marking which keys are supposed to be the ones to be consumed.

Other than that, why touch something that isn't broken?

We always can remove keys in `speced.def` - but `utils.spec` as a tiny agnostic library shouldn't introduce breakage, complexities etc